qa/tasks/cephfs/test_client_limits.py

```diff
@@ -221,6 +222,52 @@ def test_client_release_bug(self):
         self.fs.mds_asok(['session', 'evict', "%s" % mount_a_client_id])
         rproc.wait()
 
+    def test_client_blocklisted_oldest_tid(self):
+        """
+        that a client is blocklisted when its encoded session metadata exceeds the
+        configured threshold (due to ever growing `completed_requests` caused due
+        to an unidentified bug (in the client or the MDS)).
+        """
+
+        # num of requests client issues
+        max_requests = 10000
+
+        # The debug hook to inject the failure only exists in the fuse client
+        if not isinstance(self.mount_a, FuseMount):
+            self.skipTest("Require FUSE client to inject client release failure")
+
+        self.config_set('client', 'client inject fixed oldest tid', 'true')
+        self.mount_a.teardown()
+        self.mount_a.mount_wait()
+
+        self.config_set('mds', 'mds_max_completed_requests', max_requests);
+
+        # Create lots of files
+        self.mount_a.create_n_files("testdir/file1", max_requests + 100)
+
+        # Create a few files synchronously. This makes sure previous requests are completed
+        self.mount_a.create_n_files("testdir/file2", 5, True)
+
+        # Wait for the health warnings. Assume mds can handle 10 request per second at least
+        self.wait_for_health("MDS_CLIENT_OLDEST_TID", max_requests // 10, check_in_detail=str(self.mount_a.client_id))
+
+        # why a multiplier of 20, you may ask - I arrieved at this from some debugs
+        # that I put when testing the fix in a vstart cluster where its a ratio of
+        # encoded session information to the number of completed requests.
+        self.config_set('mds', 'mds_session_metadata_threshold', max_requests*20);
+
+        # Create a few more files synchronously. This would hit the session metadata threshold
+        # causing the client to get blocklisted.
+        with self.assertRaises(CommandFailedError):
+            self.mount_a.create_n_files("testdir/file2", 20, True)
+
+        self.mds_cluster.is_addr_blocklisted(self.mount_a.get_global_addr())
+
+        # reset the config
+        self.config_set('client', 'client inject fixed oldest tid', 'false')
+        self.mount_a.teardown()
+        self.mount_a.mount_wait()
```

Comment:
`self.mount_a.kill_cleanup()` would be better while teardown

Reply:
Do you mean to used kill and kill_cleanup after teardown? teardown is required since it kills the background process (if any).

Reply:
yeah i think kill and kill_cleanup would be better

Reply:
Could you explain why?

Reply:
i suspect mount to hang in case the client has been blocklisted, i know you do `teardown()` but that doesn't do force umount as the `kill_cleanup()` would do. Instead I feel a sure shot would be to `kill` and `kill_cleanup` and then `mount_wait`

Reply:
Hmm.. Agree. I'll fix this up.